### PR TITLE
Alasdair/ecom 2859 validation styles

### PR DESCRIPTION
**Pull request by AlasdairSwan**
@rlucioni @jimabramson please review this update that adds validation styling to the program creation form.

### CODE REVIEW

File: programs/static/js/models/program_model.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ define([
                         url: this.urlRoot + params,
                         // The API requires a CSRF token for all POST requests using session authentication.
                         headers: {'X-CSRFToken': $.cookie('programs_csrftoken')},
-                        contentType: patch ? 'application/merge-patch+json' : 'application/json',
+                        contentType: 'application/merge-patch+json',
```
Comment (rlucioni) [inline programs/static/js/models/program_model.js]: Why the change here? The content type on a POST should remain "application/json."

Comment (AlasdairSwan) [inline programs/static/js/models/program_model.js]: To enable PUSH. After @jimabramson's API update this is now required.

File: programs/static/js/models/program_model.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ define([
                     // XSeries is currently the only valid Program type.
                     oneOf: ['xseries']
                 },
-                organization: 'validateOrganization',
+                organizations: 'validateOrganization',
```
Comment (rlucioni) [inline programs/static/js/models/program_model.js]: Could you rename the validation method to match (i.e., `validateOrganizations`).

**Comment by AlasdairSwan** [comment]
@rlucioni updated in light of you review.

**Comment by jimabramson** [comment]
LGTM :+1:

**Comment by rlucioni** [comment]
:+1:
